### ISSUE TITLE
RuntimeError on import

### ISSUE THREAD

**Issue by jtesta** [dshikashio/Pybag#25]
I installed WinDbg on Windows 10, and I get a `RuntimeError` when importing the module:
```
C:\Users\jsmit>pip show pybag
Name: Pybag
Version: 2.2.12
[...]

C:\Users\jsmit>python
Python 3.12.7 (tags/v3.12.7:0b05ead, Oct  1 2024, 03:06:41) [MSC v.1941 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pybag
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\jsmit\AppData\Roaming\Python\Python312\site-packages\pybag\__init__.py", line 39, in <module>
    dbgdir = find_dbgdir()
             ^^^^^^^^^^^^^
  File "C:\Users\jsmit\AppData\Roaming\Python\Python312\site-packages\pybag\__init__.py", line 36, in find_dbgdir
    raise RuntimeError("Windbg install directory not found!")
RuntimeError: Windbg install directory not found!
```

**Comment by dshikashio** [comment]
Is your Windbg installation from the app store or from the WDK or?

**Comment by Algebro7** [comment]
I am getting this error as well. Windbg was installed from the app store...is that not supported?

**Comment by jtesta** [comment]
I couldn't remember if my original report was done against WinDbg from the MS app store, or from the stand-alone installer (via https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/#install-windbg-directly), so I tried it both ways (though on Windows 11 this time).  It turns out that both methods result in the same error and same stack dump.

Unless there is a workaround available, pybag seems to be critically broken at this point in time.

**Comment by dshikashio** [comment]
I never use the new version of Windbg except for TTD and in that case I always have a version from the WDK installed already. So having the app store only version installed is just something I've never tried.

After some investigation, I don't immediately see an easy way to determine installation directory for Windbg when installed from the App Store. Additionally, if you do you know the correct path, you (probably) won't have permissions to access the directory. If permissions aren't a problem, then you could set the WINDBG_DIR environment variable to the installation directory + architecture (e.g. $env:WINDBG_DIR = 'C:\Program Files\WindowsApps\Microsoft.WinDbg_1.2410.11001.0_x64__8wekyb3d8bbwe\amd64').

See here for how pybag is determining WDK installation directory - https://github.com/dshikashio/Pybag/blob/master/pybag/__init__.py

I would imagine for any WDK install, setting WINDBG_DIR should work.

It looks like BinaryNinja has a solution for this problem by doing a separate download and app "install" to a known location: https://github.com/Vector35/debugger/blob/dev/core/adapters/dbgeng/install_windbg.py

**Comment by dshikashio** [comment]
@jtesta Thanks for spending time on this!

TLDR: dbgeng.dll from the app store Windbg will work fine. I had to "side load" Windbg via the method Binary Ninja came up with to not have to play around with permissions etc.

For a variety of reasons, I created a new branch - https://github.com/dshikashio/Pybag/tree/v3. I added an entry point 'install_windbg' that will pull from the app store. Beyond that I'm trying to have a cleaner separation between dbgeng wrapping code from pybag itself. I can't guarantee I didn't break something in there if you give it a try.

At the very least, I would suggest looking at the install_windbg script for an easy way to get a usable Windbg setup. Then you can set the WINDBG_DIR environment variable appropriately and should be good to go.

Script: https://github.com/dshikashio/Pybag/blob/v3/pybag/scripts/install_windbg.py

The only other solution I've found is to just include the DLLs as part of the python package itself which I'm hesitant to do.

**Comment by dshikashio** [comment]
Thanks @jtesta for all the work to resolve this. Hopefully 2.2.13 solves the issue.
